docs/root/start/building/local_docker_build.rst

```diff
@@ -19,6 +20,12 @@ That command will take some time to run because it is compiling an Envoy binary
 
 For more information on building and different build targets, please refer to :repo:`ci/README.md`.
 
+.. warning::
```

Comment:
i think these instructions need a much stronger health warning

im uncomfortable landing this

Reply:
What do you suggest?

TBH building Envoy is an advanced topic and I wonder if we should remove it from the getting started guide entirely.

Reply:
The problem is that i ended up on this page very quickly. I really dont like these docs, but  if we remove google will just turn up the old versions and people will still follow, or  it wont, and there are no docs for building a docker image.

This was my route into envoy

- check for official packages - none
- check for docker images - no latest, unclear versioning, no info page, unclear Dockerfile source
- ok, ill build my own (as i commonly do if i dont trust existing binary channels)
- land on this page, follow instructions, both hose my (resource-constrained) machine for different reasons, swear a lot
- read source code of run_envoy_docker and swear a lot more

at this point i wrongly concluded that the project wasnt mature and/or well supported, and certainly wasnt  well documented.

thankfully, i opened a PR with the unroot as that is what i had added in my own recipe, so i gave the  project further consideration

tldr - i think it cant stay there as it is, but at this point i think building your own containers should not be regarded as an advanced task, because i think it is often necessary

Reply:
> What do you suggest?

addressing the underlying issues - eg the issues/prs that i opened.

a couple more things also i think - separating docs and docker build containers from main development/build container.

The docker build container has no need for most of the compile stuff but does need qemu, and host/root privileges. It could be very small and documented exactly what it does and how it works

The docs build doesnt need 90% of what is in the dev/build image. The current setup requires a 5GB image to run sphinx and protoc

Lastly, i guess there is a good reason for statically linking all of the compiled llvm binaries - its beyond my compiler knowledge to understand why - but that afaict is what makes that image so large

Reply:
...on further checking, maybe they are not statically linked - beyond my understanding - but they are much larger than the binaries in the distros

Reply:
As I said elsewhere, I would rather not let perfect be the enemy of the good. Can we land this overall improvement and then we can discuss what other changes you want to make? 

Personally, I think you are overstimating how important building Envoy is to the average user and I would rather it be put in some other section but we can sort that out later.

Reply:
ive strengthened the warning on the first command.

the issue with the second command should be resolved by #13618